### ISSUE TITLE
Kubernetes simple-policy tutorial inconsistent results with service on release candidate

### ISSUE THREAD

**Issue by tmjd** [projectcalico/calico#741]
tl;dr I saw inconsistent results when accessing a service IP.

This occurred when going through the tutorial on the 2.2 release candidate at v2.2/getting-started/kubernetes/tutorials/simple-policy.
I saw this behavior after:
- enabling isolation
- adding the policy to allow access

Trying to access the nginx service with `wget -q --timeout=5 nginx -O -` resulted in a successful command 3 out of 10 attempts.

I looked up the service IP address and used that to eliminate the DNS service and the results were inconsistent still.

I tried wget using the IP addresses of the 2 pods backing the service and was able to consistently access them.  So it looks like the issue is with the service IP itself.

**Comment by tmjd** [comment]
Here is some information I collected while looking at this issue.  The problem looks to be that the traffic gets NAT'ed with the src address being a tunl0 address which is not allowed by policy and is dropped.

The sequence is:
1. On node-01: Pod sends packet to service IP: src 192.168.44.200 -> dst 10.100.0.186
2. On node-01: Packet is redirected to pod IP: src 192.168.44.192 -> dst 192.168.154.193
3. On node-02: Packet received: src: 192.168.44.192 -> dst 192.168.154.193
4. *I think* that iptables is dropping the packet at this line
`[3:180] -A cali-pri-k8s_ns.policy-demo -m comment --comment "cali:s7HlNjqBTQgCRlGZ" -j DROP`

Node-01:
- has nginx pod with 192.168.154.193

Node-02
- has 'access' pod with 192.168.44.200
- tunl0 address 192.168.44.192

Nginx Service address 10.100.0.186

<details><summary>On host with pod: tcpdump -i any port 80</summary>

```
[root@k8s-node-01 ~]# tcpdump -i any port 80
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on any, link-type LINUX_SLL (Linux cooked), capture size 262144 bytes
21:52:36.858700 IP 192.168.44.192.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5683439 ecr 0,nop,wscale 7], length 0
21:52:37.895840 IP 192.168.44.192.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5684480 ecr 0,nop,wscale 7], length 0
21:52:39.944356 IP 192.168.44.192.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5686528 ecr 0,nop,wscale 7], length 0
```

</details>

<details><summary>On host with src pod: tcpdump -i any port 80</summary>

```
21:52:36.836595 IP 192.168.44.200.36528 > 10.100.0.186.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5683439 ecr 0,nop,wscale 7], length 0
21:52:36.836635 IP k8s-node-02.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5683439 ecr 0,nop,wscale 7], length 0
21:52:37.873612 IP 192.168.44.200.36528 > 10.100.0.186.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5684480 ecr 0,nop,wscale 7], length 0
21:52:37.873649 IP k8s-node-02.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5684480 ecr 0,nop,wscale 7], length 0
21:52:39.921659 IP 192.168.44.200.36528 > 10.100.0.186.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5686528 ecr 0,nop,wscale 7], length 0
21:52:39.921720 IP k8s-node-02.36528 > 192.168.154.193.http: Flags [S], seq 3569723146, win 29200, options [mss 1460,sackOK,TS val 5686528 ecr 0,nop,wscale 7], length 0
```

</details>

<details><summary>on host with pod: iptables-save -c</summary>

```
# Generated by iptables-save v1.4.21 on Tue May  2 21:57:32 2017
*raw
:PREROUTING ACCEPT [111030:155063199]
:OUTPUT ACCEPT [113050:6438602]
:cali-OUTPUT - [0:0]
:cali-PREROUTING - [0:0]
:cali-failsafe-in - [0:0]
:cali-failsafe-out - [0:0]
:cali-from-host-endpoint - [0:0]
:cali-pi-_zq488XcrF_lr5o9On9C - [0:0]
:cali-pi-k8s-policy-no-match - [0:0]
:cali-po-_zq488XcrF_lr5o9On9C - [0:0]
:cali-po-k8s-policy-no-match - [0:0]
:cali-to-host-endpoint - [0:0]
[264515:255143764] -A PREROUTING -m comment --comment "cali:6gwbT8clXdHdC1b1" -j cali-PREROUTING
[271899:17922684] -A OUTPUT -m comment --comment "cali:tVnHkvAo15HuiPy0" -j cali-OUTPUT
[271899:17922684] -A cali-OUTPUT -m comment --comment "cali:38nOqDjL6rORZtSl" -j MARK --set-xmark 0x0/0x7000000
[271899:17922684] -A cali-OUTPUT -m comment --comment "cali:mDDUhMDnNdaIUtPr" -j cali-to-host-endpoint
[0:0] -A cali-OUTPUT -m comment --comment "cali:qxtWla1G8uqJMI9B" -m mark --mark 0x1000000/0x1000000 -j ACCEPT
[264515:255143764] -A cali-PREROUTING -m comment --comment "cali:x4XbVMc5P_kNXnTy" -j MARK --set-xmark 0x0/0x7000000
[349:49149] -A cali-PREROUTING -i cali+ -m comment --comment "cali:fQeZek80kVOPa0xO" -j MARK --set-xmark 0x4000000/0x4000000
[264166:255094615] -A cali-PREROUTING -m comment --comment "cali:xp3NolkIpulCQL_G" -m mark --mark 0x0/0x4000000 -j cali-from-host-endpoint
[0:0] -A cali-PREROUTING -m comment --comment "cali:fbdE50A0BiINbNiA" -m mark --mark 0x1000000/0x1000000 -j ACCEPT
[0:0] -A cali-failsafe-in -p tcp -m comment --comment "cali:wWFQM43tJU7wwnFZ" -m multiport --dports 22 -j ACCEPT
[0:0] -A cali-failsafe-in -p udp -m comment --comment "cali:LwNV--R8MjeUYacw" -m multiport --dports 68 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:73bZKoyDfOpFwC2T" -m multiport --dports 2379 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:QMFuWo6o-d9yOpNm" -m multiport --dports 2380 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:Kup7QkrsdmfGX0uL" -m multiport --dports 4001 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:xYYr5PEqDf_Pqfkv" -m multiport --dports 7001 -j ACCEPT
[0:0] -A cali-failsafe-out -p udp -m comment --comment "cali:nbWBvu4OtudVY60Q" -m multiport --dports 53 -j ACCEPT
[0:0] -A cali-failsafe-out -p udp -m comment --comment "cali:UxFu5cDK5En6dT3Y" -m multiport --dports 67 -j ACCEPT
[0:0] -A cali-pi-_zq488XcrF_lr5o9On9C -m comment --comment "cali:S1bffVRkALV920Lz" -m set --match-set cali4-s:NBxOb5RBGENJd3bFnbDQJux src -j MARK --set-xmark 0x1000000/0x1000000
[0:0] -A cali-pi-_zq488XcrF_lr5o9On9C -m comment --comment "cali:q_WvNL6TY9qOMUxT" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-pi-k8s-policy-no-match -m comment --comment "cali:eXR8WKtGQfKPd5zm" -j MARK --set-xmark 0x2000000/0x2000000
[0:0] -A cali-pi-k8s-policy-no-match -m comment --comment "cali:J7UwAp2kUUNYDEbZ" -m mark --mark 0x2000000/0x2000000 -j RETURN
[0:0] -A cali-po-_zq488XcrF_lr5o9On9C -m comment --comment "cali:rUNpGIBr7C8VbJLH" -j MARK --set-xmark 0x1000000/0x1000000
[0:0] -A cali-po-_zq488XcrF_lr5o9On9C -m comment --comment "cali:vdGOuWqTwDyuzluu" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-po-k8s-policy-no-match -m comment --comment "cali:M1MvnGSuWnBDoJxY" -j MARK --set-xmark 0x2000000/0x2000000
[0:0] -A cali-po-k8s-policy-no-match -m comment --comment "cali:srq_4spRBeZ7r-5T" -m mark --mark 0x2000000/0x2000000 -j RETURN
COMMIT
# Completed on Tue May  2 21:57:32 2017
# Generated by iptables-save v1.4.21 on Tue May  2 21:57:32 2017
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [1:60]
:POSTROUTING ACCEPT [1:60]
:DOCKER - [0:0]
:KUBE-MARK-DROP - [0:0]
:KUBE-MARK-MASQ - [0:0]
:KUBE-NODEPORTS - [0:0]
:KUBE-POSTROUTING - [0:0]
:KUBE-SEP-4F3IDBRE6Z6F7Y23 - [0:0]
:KUBE-SEP-5XJEMYCEMS4ASSOJ - [0:0]
:KUBE-SEP-7NR2EV65N4MOMQ37 - [0:0]
:KUBE-SEP-FOGBDETOBTBXLB66 - [0:0]
:KUBE-SEP-QDN5ZB4PDR2XYQHH - [0:0]
:KUBE-SERVICES - [0:0]
:KUBE-SVC-3VXIGVIYYFN7DHDA - [0:0]
:KUBE-SVC-ERIFXISQEP7F7OF4 - [0:0]
:KUBE-SVC-NPX46M4PTMTKRN6Y - [0:0]
:KUBE-SVC-TCOU7JCQXEZGVUNU - [0:0]
:cali-OUTPUT - [0:0]
:cali-POSTROUTING - [0:0]
:cali-PREROUTING - [0:0]
:cali-fip-dnat - [0:0]
:cali-fip-snat - [0:0]
:cali-nat-outgoing - [0:0]
[235:17646] -A PREROUTING -m comment --comment "cali:6gwbT8clXdHdC1b1" -j cali-PREROUTING
[248:18616] -A PREROUTING -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
[15:816] -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
[2202:133197] -A OUTPUT -m comment --comment "cali:tVnHkvAo15HuiPy0" -j cali-OUTPUT
[2302:139242] -A OUTPUT -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
[1:84] -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
[2334:141645] -A POSTROUTING -m comment --comment "cali:O3lYWMrLQYEMJtB5" -j cali-POSTROUTING
[2686:162837] -A POSTROUTING -m comment --comment "kubernetes postrouting rules" -j KUBE-POSTROUTING
[0:0] -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
[0:0] -A DOCKER -i docker0 -j RETURN
[0:0] -A KUBE-MARK-DROP -j MARK --set-xmark 0x8000/0x8000
[3:180] -A KUBE-MARK-MASQ -j MARK --set-xmark 0x4000/0x4000
[3:180] -A KUBE-POSTROUTING -m comment --comment "kubernetes service traffic requiring SNAT" -m mark --mark 0x4000/0x4000 -j MASQUERADE
[0:0] -A KUBE-SEP-4F3IDBRE6Z6F7Y23 -s 192.168.44.194/32 -m comment --comment "policy-demo/nginx:" -j KUBE-MARK-MASQ
[0:0] -A KUBE-SEP-4F3IDBRE6Z6F7Y23 -p tcp -m comment --comment "policy-demo/nginx:" -m tcp -j DNAT --to-destination 192.168.44.194:80
[0:0] -A KUBE-SEP-5XJEMYCEMS4ASSOJ -s 192.168.154.193/32 -m comment --comment "policy-demo/nginx:" -j KUBE-MARK-MASQ
[0:0] -A KUBE-SEP-5XJEMYCEMS4ASSOJ -p tcp -m comment --comment "policy-demo/nginx:" -m tcp -j DNAT --to-destination 192.168.154.193:80
[0:0] -A KUBE-SEP-7NR2EV65N4MOMQ37 -s 192.168.44.193/32 -m comment --comment "kube-system/kube-dns:dns-tcp" -j KUBE-MARK-MASQ
[0:0] -A KUBE-SEP-7NR2EV65N4MOMQ37 -p tcp -m comment --comment "kube-system/kube-dns:dns-tcp" -m tcp -j DNAT --to-destination 192.168.44.193:53
[0:0] -A KUBE-SEP-FOGBDETOBTBXLB66 -s 192.168.44.193/32 -m comment --comment "kube-system/kube-dns:dns" -j KUBE-MARK-MASQ
[0:0] -A KUBE-SEP-FOGBDETOBTBXLB66 -p udp -m comment --comment "kube-system/kube-dns:dns" -m udp -j DNAT --to-destination 192.168.44.193:53
[0:0] -A KUBE-SEP-QDN5ZB4PDR2XYQHH -s 172.18.18.101/32 -m comment --comment "default/kubernetes:https" -j KUBE-MARK-MASQ
[3:180] -A KUBE-SEP-QDN5ZB4PDR2XYQHH -p tcp -m comment --comment "default/kubernetes:https" -m recent --set --name KUBE-SEP-QDN5ZB4PDR2XYQHH --mask 255.255.255.255 --rsource -m tcp -j DNAT --to-destination 172.18.18.101:6443
[0:0] -A KUBE-SERVICES ! -s 10.244.0.0/16 -d 10.100.0.186/32 -p tcp -m comment --comment "policy-demo/nginx: cluster IP" -m tcp --dport 80 -j KUBE-MARK-MASQ
[0:0] -A KUBE-SERVICES -d 10.100.0.186/32 -p tcp -m comment --comment "policy-demo/nginx: cluster IP" -m tcp --dport 80 -j KUBE-SVC-3VXIGVIYYFN7DHDA
[3:180] -A KUBE-SERVICES ! -s 10.244.0.0/16 -d 10.100.0.1/32 -p tcp -m comment --comment "default/kubernetes:https cluster IP" -m tcp --dport 443 -j KUBE-MARK-MASQ
[3:180] -A KUBE-SERVICES -d 10.100.0.1/32 -p tcp -m comment --comment "default/kubernetes:https cluster IP" -m tcp --dport 443 -j KUBE-SVC-NPX46M4PTMTKRN6Y
[0:0] -A KUBE-SERVICES ! -s 10.244.0.0/16 -d 10.100.0.10/32 -p udp -m comment --comment "kube-system/kube-dns:dns cluster IP" -m udp --dport 53 -j KUBE-MARK-MASQ
[0:0] -A KUBE-SERVICES -d 10.100.0.10/32 -p udp -m comment --comment "kube-system/kube-dns:dns cluster IP" -m udp --dport 53 -j KUBE-SVC-TCOU7JCQXEZGVUNU
[0:0] -A KUBE-SERVICES ! -s 10.244.0.0/16 -d 10.100.0.10/32 -p tcp -m comment --comment "kube-system/kube-dns:dns-tcp cluster IP" -m tcp --dport 53 -j KUBE-MARK-MASQ
[0:0] -A KUBE-SERVICES -d 10.100.0.10/32 -p tcp -m comment --comment "kube-system/kube-dns:dns-tcp cluster IP" -m tcp --dport 53 -j KUBE-SVC-ERIFXISQEP7F7OF4
[0:0] -A KUBE-SERVICES -m comment --comment "kubernetes service nodeports; NOTE: this must be the last rule in this chain" -m addrtype --dst-type LOCAL -j KUBE-NODEPORTS
[0:0] -A KUBE-SVC-3VXIGVIYYFN7DHDA -m comment --comment "policy-demo/nginx:" -m statistic --mode random --probability 0.50000000000 -j KUBE-SEP-5XJEMYCEMS4ASSOJ
[0:0] -A KUBE-SVC-3VXIGVIYYFN7DHDA -m comment --comment "policy-demo/nginx:" -j KUBE-SEP-4F3IDBRE6Z6F7Y23
[0:0] -A KUBE-SVC-ERIFXISQEP7F7OF4 -m comment --comment "kube-system/kube-dns:dns-tcp" -j KUBE-SEP-7NR2EV65N4MOMQ37
[3:180] -A KUBE-SVC-NPX46M4PTMTKRN6Y -m comment --comment "default/kubernetes:https" -m recent --rcheck --seconds 10800 --reap --name KUBE-SEP-QDN5ZB4PDR2XYQHH --mask 255.255.255.255 --rsource -j KUBE-SEP-QDN5ZB4PDR2XYQHH
[0:0] -A KUBE-SVC-NPX46M4PTMTKRN6Y -m comment --comment "default/kubernetes:https" -j KUBE-SEP-QDN5ZB4PDR2XYQHH
[0:0] -A KUBE-SVC-TCOU7JCQXEZGVUNU -m comment --comment "kube-system/kube-dns:dns" -j KUBE-SEP-FOGBDETOBTBXLB66
[2202:133197] -A cali-OUTPUT -m comment --comment "cali:GBTAv2p5CwevEyJm" -j cali-fip-dnat
[2334:141645] -A cali-POSTROUTING -m comment --comment "cali:Z-c7XtVd2Bq7s_hA" -j cali-fip-snat
[2334:141645] -A cali-POSTROUTING -m comment --comment "cali:nYKhEzDlr11Jccal" -j cali-nat-outgoing
[0:0] -A cali-POSTROUTING -o tunl0 -m comment --comment "cali:JHlpT-eSqR1TvyYm" -m addrtype ! --src-type LOCAL --limit-iface-out -m addrtype --src-type LOCAL -j MASQUERADE
[235:17646] -A cali-PREROUTING -m comment --comment "cali:r6XmIziWUJsdOK6Z" -j cali-fip-dnat
[0:0] -A cali-nat-outgoing -m comment --comment "cali:Wd76s91357Uv7N3v" -m set --match-set cali4-masq-ipam-pools src -m set ! --match-set cali4-all-ipam-pools dst -j MASQUERADE
COMMIT
# Completed on Tue May  2 21:57:32 2017
# Generated by iptables-save v1.4.21 on Tue May  2 21:57:32 2017
*filter
:INPUT ACCEPT [324:38030]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [313:24718]
:DOCKER - [0:0]
:DOCKER-ISOLATION - [0:0]
:KUBE-FIREWALL - [0:0]
:KUBE-SERVICES - [0:0]
:cali-FORWARD - [0:0]
:cali-INPUT - [0:0]
:cali-OUTPUT - [0:0]
:cali-failsafe-in - [0:0]
:cali-failsafe-out - [0:0]
:cali-from-host-endpoint - [0:0]
:cali-from-wl-dispatch - [0:0]
:cali-fw-calibee8ec819d7 - [0:0]
:cali-pi-_zq488XcrF_lr5o9On9C - [0:0]
:cali-pi-k8s-policy-no-match - [0:0]
:cali-po-_zq488XcrF_lr5o9On9C - [0:0]
:cali-po-k8s-policy-no-match - [0:0]
:cali-pri-k8s_ns.policy-demo - [0:0]
:cali-pro-k8s_ns.policy-demo - [0:0]
:cali-to-host-endpoint - [0:0]
:cali-to-wl-dispatch - [0:0]
:cali-tw-calibee8ec819d7 - [0:0]
:cali-wl-to-host - [0:0]
[9157:1093993] -A INPUT -m comment --comment "cali:Cz_u1IQiXIMmKD4c" -j cali-INPUT
[9157:1093993] -A INPUT -j KUBE-FIREWALL
[3:180] -A FORWARD -m comment --comment "cali:wUHhoiAYhphO9Mso" -j cali-FORWARD
[0:0] -A FORWARD -j DOCKER-ISOLATION
[0:0] -A FORWARD -o docker0 -j DOCKER
[0:0] -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
[0:0] -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
[0:0] -A FORWARD -i docker0 -o docker0 -j ACCEPT
[9520:758052] -A OUTPUT -m comment --comment "cali:tVnHkvAo15HuiPy0" -j cali-OUTPUT
[9520:758052] -A OUTPUT -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
[9520:758052] -A OUTPUT -j KUBE-FIREWALL
[0:0] -A DOCKER-ISOLATION -j RETURN
[0:0] -A KUBE-FIREWALL -m comment --comment "kubernetes firewall for dropping marked packets" -m mark --mark 0x8000/0x8000 -j DROP
[0:0] -A cali-FORWARD -m comment --comment "cali:jxvuJjmmRV135nVu" -m mark --mark 0x1000000/0x1000000 -m conntrack --ctstate UNTRACKED -j ACCEPT
[0:0] -A cali-FORWARD -i cali+ -m comment --comment "cali:nu_3aWP3DUkeeFF6" -j cali-from-wl-dispatch
[3:180] -A cali-FORWARD -o cali+ -m comment --comment "cali:DjrV_uMYqr-g4joA" -j cali-to-wl-dispatch
[0:0] -A cali-FORWARD -i cali+ -m comment --comment "cali:Hl34eZwIcbzmic3y" -j ACCEPT
[0:0] -A cali-FORWARD -o cali+ -m comment --comment "cali:O17zRKq2dvqwJKGA" -j ACCEPT
[0:0] -A cali-FORWARD -m comment --comment "cali:aTQofb9V5IPBvpDr" -j MARK --set-xmark 0x0/0x7000000
[0:0] -A cali-FORWARD -m comment --comment "cali:yl6jfcAHxkOSlAV7" -j cali-from-host-endpoint
[0:0] -A cali-FORWARD -m comment --comment "cali:zA6HyaP1JlANkvKN" -j cali-to-host-endpoint
[0:0] -A cali-FORWARD -m comment --comment "cali:xYGCuGpZAkaFt1KN" -m comment --comment "Host endpoint policy accepted packet." -m mark --mark 0x1000000/0x1000000 -j ACCEPT
[0:0] -A cali-INPUT -m comment --comment "cali:46gVAqzWLjH8U4O2" -m mark --mark 0x1000000/0x1000000 -m conntrack --ctstate UNTRACKED -j ACCEPT
[0:0] -A cali-INPUT -p ipv4 -m comment --comment "cali:aR-jVK2O7VHKhzJt" -m comment --comment "Drop IPIP packets from non-Calico hosts" -m set ! --match-set cali4-all-hosts src -j DROP
[0:0] -A cali-INPUT -i cali+ -m comment --comment "cali:gVjPMLEBhhpD4I0U" -g cali-wl-to-host
[9157:1093993] -A cali-INPUT -m comment --comment "cali:THWbvZjtqLQwsqiH" -j MARK --set-xmark 0x0/0x7000000
[9157:1093993] -A cali-INPUT -m comment --comment "cali:6wnP2LUCc7lG2gnl" -j cali-from-host-endpoint
[0:0] -A cali-INPUT -m comment --comment "cali:I3YjhDkapGtZ-w-C" -m comment --comment "Host endpoint policy accepted packet." -m mark --mark 0x1000000/0x1000000 -j ACCEPT
[0:0] -A cali-OUTPUT -m comment --comment "cali:FwFFCT8uDthhfgS7" -m mark --mark 0x1000000/0x1000000 -m conntrack --ctstate UNTRACKED -j ACCEPT
[0:0] -A cali-OUTPUT -o cali+ -m comment --comment "cali:lE9pRQNw1a_fJ2-L" -j RETURN
[9520:758052] -A cali-OUTPUT -m comment --comment "cali:kXSia9_8D_I9Mx8M" -j MARK --set-xmark 0x0/0x7000000
[9520:758052] -A cali-OUTPUT -m comment --comment "cali:xuyU_DgoL_xoueJt" -j cali-to-host-endpoint
[0:0] -A cali-OUTPUT -m comment --comment "cali:-KZpg9OTpqQcNRfw" -m comment --comment "Host endpoint policy accepted packet." -m mark --mark 0x1000000/0x1000000 -j ACCEPT
[0:0] -A cali-failsafe-in -p tcp -m comment --comment "cali:wWFQM43tJU7wwnFZ" -m multiport --dports 22 -j ACCEPT
[0:0] -A cali-failsafe-in -p udp -m comment --comment "cali:LwNV--R8MjeUYacw" -m multiport --dports 68 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:73bZKoyDfOpFwC2T" -m multiport --dports 2379 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:QMFuWo6o-d9yOpNm" -m multiport --dports 2380 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:Kup7QkrsdmfGX0uL" -m multiport --dports 4001 -j ACCEPT
[0:0] -A cali-failsafe-out -p tcp -m comment --comment "cali:xYYr5PEqDf_Pqfkv" -m multiport --dports 7001 -j ACCEPT
[0:0] -A cali-failsafe-out -p udp -m comment --comment "cali:nbWBvu4OtudVY60Q" -m multiport --dports 53 -j ACCEPT
[0:0] -A cali-failsafe-out -p udp -m comment --comment "cali:UxFu5cDK5En6dT3Y" -m multiport --dports 67 -j ACCEPT
[0:0] -A cali-from-wl-dispatch -i calibee8ec819d7 -m comment --comment "cali:PD2a5EcdIzU3JBZH" -g cali-fw-calibee8ec819d7
[0:0] -A cali-from-wl-dispatch -m comment --comment "cali:VyT-qFCixhh_6X4y" -m comment --comment "Unknown interface" -j DROP
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:iWd5DxiGldLtJUDw" -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:r2_Aao1MDUZIc0Kd" -m conntrack --ctstate INVALID -j DROP
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:hzNMvVmDbVqoOY-o" -j MARK --set-xmark 0x0/0x1000000
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:FeUSjsLsGj8OuXnX" -m comment --comment "Start of policies" -j MARK --set-xmark 0x0/0x2000000
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:Ni5eyZUVQAw0Qfq0" -m mark --mark 0x0/0x2000000 -j cali-po-_zq488XcrF_lr5o9On9C
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:Aq0-38S_3PEC_cr_" -m comment --comment "Return if policy accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:ihLmpRrMNTqktVt4" -m mark --mark 0x0/0x2000000 -j cali-po-k8s-policy-no-match
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:xz9i_JXgz0-6R-Zr" -m comment --comment "Return if policy accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:Ocaiodi14bFx8b5t" -m comment --comment "Drop if no policies passed packet" -m mark --mark 0x0/0x2000000 -j DROP
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:_bhj7sVpxwm7dBUc" -j cali-pro-k8s_ns.policy-demo
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:l-bd7EWzGV3xVs5o" -m comment --comment "Return if profile accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-fw-calibee8ec819d7 -m comment --comment "cali:hvwmjdHIQb_VPR9E" -m comment --comment "Drop if no profiles matched" -j DROP
[0:0] -A cali-pi-_zq488XcrF_lr5o9On9C -m comment --comment "cali:S1bffVRkALV920Lz" -m set --match-set cali4-s:NBxOb5RBGENJd3bFnbDQJux src -j MARK --set-xmark 0x1000000/0x1000000
[0:0] -A cali-pi-_zq488XcrF_lr5o9On9C -m comment --comment "cali:q_WvNL6TY9qOMUxT" -m mark --mark 0x1000000/0x1000000 -j RETURN
[3:180] -A cali-pi-k8s-policy-no-match -m comment --comment "cali:eXR8WKtGQfKPd5zm" -j MARK --set-xmark 0x2000000/0x2000000
[3:180] -A cali-pi-k8s-policy-no-match -m comment --comment "cali:J7UwAp2kUUNYDEbZ" -m mark --mark 0x2000000/0x2000000 -j RETURN
[0:0] -A cali-po-_zq488XcrF_lr5o9On9C -m comment --comment "cali:rUNpGIBr7C8VbJLH" -j MARK --set-xmark 0x1000000/0x1000000
[0:0] -A cali-po-_zq488XcrF_lr5o9On9C -m comment --comment "cali:vdGOuWqTwDyuzluu" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-po-k8s-policy-no-match -m comment --comment "cali:M1MvnGSuWnBDoJxY" -j MARK --set-xmark 0x2000000/0x2000000
[0:0] -A cali-po-k8s-policy-no-match -m comment --comment "cali:srq_4spRBeZ7r-5T" -m mark --mark 0x2000000/0x2000000 -j RETURN
[3:180] -A cali-pri-k8s_ns.policy-demo -m comment --comment "cali:s7HlNjqBTQgCRlGZ" -j DROP
[0:0] -A cali-pro-k8s_ns.policy-demo -m comment --comment "cali:gWOIaukdJxJtEFRR" -j MARK --set-xmark 0x1000000/0x1000000
[0:0] -A cali-pro-k8s_ns.policy-demo -m comment --comment "cali:joKnmX34K7aMT0zl" -m mark --mark 0x1000000/0x1000000 -j RETURN
[3:180] -A cali-to-wl-dispatch -o calibee8ec819d7 -m comment --comment "cali:TadK-Z6J6QyPn7uB" -g cali-tw-calibee8ec819d7
[0:0] -A cali-to-wl-dispatch -m comment --comment "cali:fOoWXYcjtWABvd7W" -m comment --comment "Unknown interface" -j DROP
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:iOLE3FSKmMOwte1f" -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:VfkCQ7ji8ZihD9Ki" -m conntrack --ctstate INVALID -j DROP
[3:180] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:TBqGouPig_91nmzJ" -j MARK --set-xmark 0x0/0x1000000
[3:180] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:pxkHj483uOJYIUla" -m comment --comment "Start of policies" -j MARK --set-xmark 0x0/0x2000000
[3:180] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:Ui15Gq6LYJLuLGmM" -m mark --mark 0x0/0x2000000 -j cali-pi-_zq488XcrF_lr5o9On9C
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:86hWGPArjkqJzcoL" -m comment --comment "Return if policy accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[3:180] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:k8sHjb0dxaPK06le" -m mark --mark 0x0/0x2000000 -j cali-pi-k8s-policy-no-match
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:xTZsn2GS57-eELyY" -m comment --comment "Return if policy accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:U8_7MGoO7-jnAhJT" -m comment --comment "Drop if no policies passed packet" -m mark --mark 0x0/0x2000000 -j DROP
[3:180] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:GHq6isNn4WDwFxmj" -j cali-pri-k8s_ns.policy-demo
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:ydveQmg1B1XNSVes" -m comment --comment "Return if profile accepted" -m mark --mark 0x1000000/0x1000000 -j RETURN
[0:0] -A cali-tw-calibee8ec819d7 -m comment --comment "cali:NunOkQU6xgLUXOel" -m comment --comment "Drop if no profiles matched" -j DROP
[0:0] -A cali-wl-to-host -p udp -m comment --comment "cali:aEOMPPLgak2S0Lxs" -m multiport --sports 68 -m multiport --dports 67 -j ACCEPT
[0:0] -A cali-wl-to-host -p udp -m comment --comment "cali:SzR8ejPiuXtFMS8B" -m multiport --dports 53 -j ACCEPT
[0:0] -A cali-wl-to-host -m comment --comment "cali:MEmlbCdco0Fefcrw" -j cali-from-wl-dispatch
[0:0] -A cali-wl-to-host -m comment --comment "cali:LZBoXHDOlr3ok4R3" -m comment --comment "Configured DefaultEndpointToHostAction" -j ACCEPT
COMMIT
# Completed on Tue May  2 21:57:32 2017
```

</details>

**Comment by caseydavenport** [comment]
@tmjd are you running this cluster with `ipip` enabled? Do you see the traffic being encapsulated and taking tunl0? or just leaving the host with tunl0's address?

If so, do you see it unencap'd on the other side?

**Comment by tmjd** [comment]
I didn't look for tunnel'ed traffic.  I think the issue is that the `cluster-cidr` does not match the pool CIDR.  The `cluster-cidr` was changed with some KDD changes so they currently don't match.  I'm going to test that fixing the `cluster-cidr` fixes the problem.

**Comment by caseydavenport** [comment]
Might be related to this: https://github.com/projectcalico/calico/issues/683

**Comment by tmjd** [comment]
Fixing the `cluster-cidr` fixes this issue. I'll get the PR up tomorrow for this.
